app/src/controllers/remoteHost.js

```diff
@@ -0,0 +1,44 @@
+import { generateHash, compareWithHash } from '../utils/bcrypt.js';
+import { dbAddAPIKey, dbGetAPIKeyList, dbChangeAPIKeyName } from '../db/queries.js';
+import generateAPIKey from '../utils/generateAPIKey.js';
+
+const addAPIKey = async (req, res, next) => {
+    try {
+        const apiKey = generateAPIKey();
+        const hash = await generateHash(apiKey);
+
+        const prefix = apiKey.slice(0, 8);
+
+        let data = await dbAddAPIKey(hash, prefix);
+        res.status(201).send({apiKey: apiKey, id: data.id, prefix: prefix});
+    } catch(error) {
+        next(error);
+    }
+};
+
+const addName = async (req, res, next) => {
+    try {
+        const id = req.params.id;
+        console.log(id);
+        const {name} = req.body;
+        await dbChangeAPIKeyName(name, id);
+        res.status(200).send();
+    } catch(error) {
+        next(error);
+    }
+}
+
+const verifyAPIKey = async (req, res, next) => {
+    try {
+        const apiKey = req;
+        const apiKeyList = await dbGetAPIKeyList();
+
+        const apiKeyCheck = apiKeyList.some(key => compareWithHash(apiKey, key.api_key_hash));
```

Comment:
Just realised after merging this that I'm not sure this method will work. `compareWithHash` is `async` so will return a promise which (I think) will always evaluate as `true`. Think something like `Promise.race` will need to be used here.

Reply:
Good catch; I think you're right. I'll work on this after our morning scrum 👍